aws/cloudformation/drone-stack.yml

```diff
@@ -0,0 +1,367 @@
+Description: 'AWS CloudFormation template for running Drone.io on ECS. Recreates these steps: https://docs.drone.io/installation/github/multi-machine/'
+Parameters:
+  VpcId:
+    Type: String
+    Description: VpcId
+  SubnetIds:
+    Type: CommaDelimitedList
+    Description: Comma separated list of existing VPC Subnet Ids where ECS instances will run
+  AZs:
+    Type: CommaDelimitedList
+    Description: Comma separated list of availability zones where drone master ECS instances will run
+  NumWorkers:
+    Type: Number
+    Description: Initial number of drone workers (also the number of concurrent builds which can run)
+    Default: 2
+Resources:
+# ELB resources:
+# ELB / Target Group / Listener / ELB Security Group
+
+  ElbSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ELB Allowed Ports
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+          CidrIp: 0.0.0.0/0
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - ALB SecurityGroup"
+
+  DroneMasterTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      Name: !Sub "TargetGroup-${AWS::StackName}"
+      VpcId: !Ref VpcId
+      Port: 80
+      Protocol: HTTP
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - TargetGroup"
+
+  DroneMasterELB:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Name: !Sub "ELB-${AWS::StackName}"
+      SecurityGroups:
+        - !Ref ElbSecurityGroup
+      Subnets: !Ref SubnetIds
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - ELB"
+
+  LoadBalancerListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      LoadBalancerArn: !Ref DroneMasterELB
+      Port: 80
+      Protocol: HTTP
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref DroneMasterTargetGroup
+
+# Drone Master resources:
+# Security Group / EC2 Launch Template / EC2 Autoscaling Group
+# ECS cluster / ECS service / ECS task definition
+
+  DroneMasterEcsSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ECS Allowed Ports
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        -
+          IpProtocol: tcp
+          FromPort: 1
+          ToPort: 65535
+          SourceSecurityGroupId: !Ref ElbSecurityGroup
+        -
+          IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+
+  DroneMasterECSCluster:
+    Type: AWS::ECS::Cluster
+
+  DroneMasterLaunchTemplate:
+    Type: AWS::EC2::LaunchTemplate
+    Properties:
+      LaunchTemplateName: !Sub "DroneMasterLaunchTemplate-${AWS::StackName}"
+      LaunchTemplateData: 
+        IamInstanceProfile: 
+          Name: ecsInstanceRole
+        KeyName: !Sub "winter-dev-${AWS::Region}"
```

Comment:
Does this SSH `KeyName` need to be changed?

Reply:
I'm going to leave this as a TODO (unless going forward with this key is a problem). Ideally we could SSH from gateway, and then we can remove this key.